src/lib/spinel/radio_spinel.cpp

```diff
@@ -198,6 +200,13 @@ otError RadioSpinel::CheckSpinelVersion(void)
     {
         LogCrit("Spinel version mismatch - Posix:%d.%d, RCP:%d.%d", SPINEL_PROTOCOL_VERSION_THREAD_MAJOR,
                 SPINEL_PROTOCOL_VERSION_THREAD_MINOR, versionMajor, versionMinor);
+
+#if OPENTHREAD_SPINEL_CONFIG_VENDOR_HOOK_ENABLE
+        if (mVendorHandleErrorCallback)
+        {
+            mVendorHandleErrorCallback(OT_EXIT_RADIO_SPINEL_INCOMPATIBLE, mVendorHandleErrorContext);
+        }
+#endif
         DieNow(OT_EXIT_RADIO_SPINEL_INCOMPATIBLE);
```

Comment:
This code pattern (including the line of `DieNow`) is common in this PR. Can we write a util function to encapsulate it?